qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -184,12 +185,14 @@ def use_nplc_for_high_resolution_adc(
                 ``Integration time = n / power line frequency``.
 
                 n=1 to 100. Default setting is 1 (if `None` is passed).
-                (For more info see Table 4-21.)
+                (For more info see Table 4-21.).  Note that the integration
+                time will not be updated if a non-integer value is written
+                to the B1500.
         """
         self._setup_integration_time(
             adc_type=constants.AIT.Type.HIGH_RESOLUTION,
             mode=constants.AIT.Mode.NPLC,
-            coeff=n
+            coeff=int(n)
```

Comment:
and for simplicity this conversion will be done in `_setup_integration_time`, not in every place where it's is called.